src/mmda/parsers/grobid_full_text_parser.py

```diff
@@ -0,0 +1,112 @@
+"""
+
+@geli-gel
+
+"""
+from grobid_client.grobid_client import GrobidClient
+from typing import Optional, List
+
+import io
+import os
+import xml.etree.ElementTree as et
+
+from mmda.parsers.parser import Parser
+from mmda.types.annotation import BoxGroup, Box
+from mmda.types.document import Document
+from mmda.types.names import PagesField, RowsField, TokensField
+
+REQUIRED_DOCUMENT_FIELDS = [PagesField, RowsField, TokensField]
+NS = {"tei": "http://www.tei-c.org/ns/1.0"}
+
+
+class GrobidFullTextParser(Parser):
+    """Grobid parser that uses grobid python client to convert grobid XML 
+    TEI coordinates into MMDA SpanGroups.
+    """
+
+    def __init__(self, config_path: str = "grobid.config"):
+
+        self.client = GrobidClient(config_path=config_path)
+
+
+    def parse(self, input_pdf_path: str, doc: Document, xml_out_dir: Optional[str] = None) -> Document:
+        
+        assert doc.symbols != ""
+        for field in REQUIRED_DOCUMENT_FIELDS:
+            assert field in doc.fields
+
+        (_, _, xml) = self.client.process_pdf(
+                "processFulltextDocument",
+                input_pdf_path,
+                generateIDs=False,
+                consolidate_header=False,
+                consolidate_citations=False,
+                include_raw_citations=False,
+                include_raw_affiliations=False,
+                tei_coordinates=True,
+                segment_sentences=True
+            )
+        if xml_out_dir:
+            os.makedirs(xml_out_dir, exist_ok=True)
+            xmlfile = os.path.join(xml_out_dir, os.path.basename(input_pdf_path).replace('.pdf', '.xml'))
+            with open(xmlfile, 'w') as f_out:
+                f_out.write(xml)
+
+        self._parse_xml_onto_doc(xml, doc)
+        return doc
+
+    def _parse_xml_onto_doc(self, xml: str, doc: Document) -> Document:
+        xml_root = et.parse(io.StringIO(xml)).getroot()
+
+        bib_entries = self._get_grobid_bib_box_groups(xml_root)
+        
+        # if/when adding in relations between mention sources and bib targets, we'll have to 
+        #  re-map the source and target IDs onto their new MMDA SpanGroup IDs
+        doc.annotate(
+            bib_entries=bib_entries
+            )
+
+        return doc
+
+    def _xml_coords_to_boxes(self, coords_attribute: str, page_sizes: dict):
```

Comment:
is there a way to test that this works as expected? Like, that this conversion from Grobid's output gives us something close to what we get via the mmda bib detector, which (I believe) follows a different path to boxes?

you mention this as work still to come:
>use the SpanGroup span boxes which essentially consolidate the grobid boxes which are drawn on a per-line basis (whereas the spangroup span boxes are drawn on a per text block basis, more similar to the bib detector we are using grobid in place of

I'm not understanding yet how this is something that still needs to be done...can you explain, or annotate the code somehow so it is clearer what step might still be missing? or is there a way a test could clarify what this means?

Reply:
I meant that as a note in regard to the overarching ticket, that this PR represents basically the first half of the work -- getting the raw grobid bibs into MMDA format - so our work in this repo is essentially finished for the bib detection replacement work. In the spp-grobid-api is where I was imagining the SpanGroup span altering code would live.

Reply:
> is there a way to test that this works as expected? Like, that this conversion from Grobid's output gives us something close to what we get via the mmda bib detector, which (I believe) follows a different path to boxes?

Once we have this code plus the other stuff in spp-grobid would be the better time to test, but as a super simple test I included a small sample in the python notebook showing the text extracted from the grobid boxes:
![image](https://user-images.githubusercontent.com/7800734/223845518-fbf97a32-0653-4832-8c18-71e84fdd5d8e.png)